source/common/common/utility.cc

```diff
@@ -486,6 +486,33 @@ std::string StringUtil::removeCharacters(const absl::string_view& str,
   return absl::StrJoin(pieces, "");
 }
 
+absl::optional<uint64_t> StringUtil::readAndRemoveLeadingDigits(absl::string_view& s) {
+  absl::optional<uint64_t> output;
+  const char* p = s.data();
+  const char* limit = p + s.size();
+  uint64_t val = 0;
+  while (p < limit) {
```

Comment:
I think this would be slightly clearer as:
size_t digits_read = 0;
for (; digits_read < s.size(); ++digits_read) {
 const char c = s[digits_read];
...
s.remove_prefix(digits_read)

If you find the current approach more readable, ignore this comment.